metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -228,8 +228,16 @@ func (m *metricHints) getHostsWithPort(hints common.MapStr, port int, noPort boo
 	return result, true
 }
 
-// getPortName returns the port name defined in a host hint ie: for ${data.host}:%{data.ports.web} returns web
+// getPortName returns the port name defined in a host hint ie: for ${data.host}:${data.ports.web} returns web
 func (m *metricHints) getPortName(h string) string {
+
+	re := regexp.MustCompile(`\${data.ports.*\}`)
```

Comment:
Nit. This regexp could be compiled to a package variable so it doesn't need to be re-compiled for each port.